bulletin_board/server/app/commands/end_vote.rb

```diff
@@ -24,13 +24,13 @@ def initialize(authority, message_id, signed_data)
   def call
     return broadcast(:invalid, error) unless
       valid_log_entry?("end_vote")
+    return broadcast(:invalid, error) unless
+      valid_step?(election.vote?) &&
+      valid_client?(authority.authority? && election.authority == authority) &&
+      valid_author?(message_identifier.from_authority?)
 
     election.with_lock do
-      return broadcast(:invalid, error) unless
-        valid_step?(election.vote?) &&
-        valid_client?(authority.authority? && election.authority == authority) &&
-        valid_author?(message_identifier.from_authority?) &&
-        process_message
+      return broadcast(:invalid, error) unless process_message # rubocop:disable Rails/TransactionExitStatement
```

Comment:
Note to reviewer: I had to leave this exit in the lock, as this method (`process_message`) is actually leaving the `election` dirty (without saving the model) 

https://github.com/decidim/decidim-bulletin-board/blob/43b5d84630fb974c4e7a8d5cbfd93eb9b7b22dac/bulletin_board/server/app/commands/concerns/log_entry_command.rb

This generated an exception: 

>       RuntimeError:
>         Locking a record with unpersisted changes is not supported. Use `save` to persist the changes, or `reload` to discard them explicitly.